src/utils/watch/fastify.ts

```diff
@@ -0,0 +1,55 @@
+import C from 'ansi-colors'
+import { FastifyInstance } from 'fastify'
+import { start, stop } from '../start'
+import { TIMEOUT } from './constants'
+import { GRACEFUL_SHUTDOWN, READY } from './events'
+
+function exit (this: any): void {
+  if (this as boolean) { console.log(C.red('[fastify-cli] process forced end')) }
+  process.exit(1)
+}
+
+// shared instance
+let fastify: null | FastifyInstance = null
+
+process.on('message', function (event) {
+  console.log('message', event)
+  if (event === GRACEFUL_SHUTDOWN) {
```

Comment:
would https://www.npmjs.com/package/close-with-grace be useful here?

Reply:
Definitively a better alternative